File: algorithms/javascript/divide.js
let totalInput = ''
process.stdin.resume()
process.stdin.setEncoding('ascii')
process.stdin.on('data', (input) => {
  totalInput += input
})

process.stdin.on('end', () => {
  processData(totalInput)
})

const processData = (input) => {
  const inArr = input.split(' ').map(Number)
  let numerator = inArr[0]
  const denominator = inArr[1]
  let divisor = 1
  let n = numerator
  for (; n < denominator; n = n + numerator, divisor++) {}
  let remainder = 0
  if (n !== denominator) {
    divisor--
    remainder = denominator - (n - numerator)
  }
  if (remainder) {
    console.log(divisor + ' Remainder ' + remainder)
  } else {
    console.log(divisor)
  }
}

File: algorithms/go/divide.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	numbers, _ := reader.ReadString('\n')
	numbers = strings.TrimSpace(numbers)
	numArray := strings.Split(numbers, " ")
	strN, strD := numArray[0], strings.TrimSpace(numArray[1])
	numerator, _ := strconv.ParseInt(strN, 10, 32)
	denominator, _ := strconv.ParseInt(strD, 10, 32)
	divisor, num := 1, numerator
	for ; num < denominator; divisor++ {
		num += numerator
	}
	if num != denominator {
		divisor--
		remainder := denominator - (num - numerator)
		fmt.Println(divisor, " Remainder ", remainder)
	} else {
		fmt.Println(divisor)
	}
}
